src/CrectoAdmin/auto_resource.cr

```diff
@@ -0,0 +1,188 @@
+module CrectoAdmin
+  class Column
+    property column_name : String
+    property column_key : String
+    property extra : String
+    property data_type : String
+
+    def initialize(@column_name, @column_key, @extra, @data_type)
+    end
+
+    def is_created_at?
+      column_name = @column_name.downcase
+      data_type = @data_type.downcase
+      (column_name == "created_at") && (data_type == "datetime")
+    end
+
+    def is_updated_at?
+      column_name = @column_name.downcase
+      data_type = @data_type.downcase
+      (column_name == "updated_at") && (data_type == "datetime")
+    end
+
+    def column_type
+      t = @data_type.upcase
+      return "Bool" if t.includes? "BOOL"
+      return "Int64" if t.includes?("INT") || t.includes?("SERIAL")
+      return "Float64" if t.includes?("DEC") || t.includes?("FLOAT") || t.includes?("DOUBLE")
+      return "Time" if t.includes? "DATETIME"
+      return "String" if t.includes?("CHAR") || t.includes?("TEXT")
+      nil
+    end
+  end
+
+  class Table
+    property db_name : String
+    property table_name : String
+    property columns : Array(CrectoAdmin::Column)
+    property class_name : String
+
+    def initialize(@db_name, @table_name)
+      @columns = [] of CrectoAdmin::Column
+      @class_name = "#{table_name[0].upcase}#{table_name[1..-1]}"
+    end
+
+    def has_created_at?
+      @columns.each do |column|
+        return true if column.is_created_at?
+      end
+      false
+    end
+
+    def has_updated_at?
+      @columns.each do |column|
+        return true if column.is_updated_at?
+      end
+      false
+    end
+
+    def primary_key_column
+      pri_columns = [] of CrectoAdmin::Column
+      @columns.each do |column|
+        column_key = column.column_key.downcase
+        pri_columns << column if column_key.includes?("pri")
+      end
+      return nil if pri_columns.empty?
+      pri_columns.each do |column|
+        return column if column.column_name.includes? "id"
+      end
+      pri_columns[0]
+    end
+
+    def has_primary_key_auto?
+      pri_column = primary_key_column
+      return false if pri_column.nil?
+      pri_column = pri_column.as(CrectoAdmin::Column)
+      pri_column.extra.includes?("auto_increment")
+    end
+
+    def has_more_columns?
+      return false if primary_key_column.nil?
+      @columns.each do |column|
+        next if column == primary_key_column
+        next if column.column_type.nil?
+        return true
+      end
+      false
+    end
+  end
+
+  def self.build_header(uri)
```

Comment:
Building strings is fine, just so you are aware ECR templates can also be used to generate files: https://crystal-lang.org/api/0.26.0/ECR.html

Reply:
Oh, I did not realize that. Yeah, that will be much better to generate crystal code. I will definitely change to that.